internal/service/iam/instance_profile.go

```diff
@@ -117,6 +130,21 @@ func resourceInstanceProfileCreate(d *schema.ResourceData, meta interface{}) err
 		return fmt.Errorf("timed out while waiting for instance profile %s: %w", name, err)
 	}
 
+	// Some partitions (i.e., ISO) may not support tag-on-create, attempt tag after create
+	if request.Tags == nil && len(tags) > 0 {
+		err := instanceProfileUpdateTags(conn, d.Id(), nil, tags)
+
+		// If default tags only, log and continue. Otherwise, error.
+		if v, ok := d.GetOk("tags"); (!ok || len(v.(map[string]interface{})) == 0) && verify.CheckISOErrorTagsUnsupported(err) {
+			log.Printf("[WARN] failed adding tags after create for IAM Instance Profile (%s): %s", d.Id(), err)
+			return resourceInstanceProfileUpdate(d, meta)
+		}
+
+		if err != nil {
+			return fmt.Errorf("error updating tags for IAM Instance Profile (%s): %w", d.Id(), err)
```

Comment:
Should this error message reference adding tags after creation? e.g.

```suggestion
			return fmt.Errorf("error adding tags after create for IAM Instance Profile (%s): %w", d.Id(), err)
```

Reply:
Nice catch 👍